packages/core/src/node/cli.ts

```diff
@@ -25,32 +25,47 @@ export const CliContribution = Symbol('CliContribution');
  * Call back for extension to contribute options to the cli.
  */
 export interface CliContribution {
-    configure(conf: yargs.Argv): void;
+    /**
+     * Configure the `yargs.Argv` parser with your options.
+     */
+    configure(conf: yargs.Argv): MaybePromise<void>;
+    /**
+     * Fetch the parsed options.
+     */
     setArguments(args: yargs.Arguments): MaybePromise<void>;
 }
 
 @injectable()
 export class CliManager {
 
-    constructor(@inject(ContributionProvider) @named(CliContribution)
-    protected readonly contributionsProvider: ContributionProvider<CliContribution>) { }
+    protected readonly parser: yargs.Argv;
 
-    async initializeCli(argv: string[]): Promise<void> {
+    constructor(
+        @inject(ContributionProvider) @named(CliContribution)
+        protected readonly contributionsProvider: ContributionProvider<CliContribution>,
+    ) {
         const pack = require('../../package.json');
-        const version = pack.version;
-        const command = yargs.version(version);
-        command.exitProcess(this.isExit());
-        for (const contrib of this.contributionsProvider.getContributions()) {
-            contrib.configure(command);
-        }
-        const args = command
+        this.parser = yargs
```

Comment:
Could you elaborate on changes in this file?

btw you can use `postConstruct` annotation to avoid constructor injection

Reply:
Also see: https://github.com/eclipse-theia/theia/pull/7964#discussion_r439269907

Reply:
I did not get the response there, on master we don't use construction injection. Using the construction injection is breaking API.

Reply:
Construction injection is used on this class on master. Although you are right and these changes are left overs.